tests/e2e/test_developer_journey.py

```diff
@@ -144,4 +144,24 @@ def test_create_and_preview_schema(
         tasklist_page.collection_status_box.filter(has=tasklist_page.page.get_by_text("In progress"))
     ).to_be_visible()
     expect(tasklist_page.submit_button).to_be_enabled()
-    tasklist_page.click_submit_collection()
+    confirmation_page = tasklist_page.click_submit_collection()
+    collection_reference = confirmation_page.collection_reference.inner_text()
+
+    # Go back to schema detail page
+    all_grants_page = AllGrantsPage(page, domain)
+    all_grants_page.navigate()
+    grant_dashboard_page = all_grants_page.click_grant(new_grant_name)
+    developers_page = grant_dashboard_page.click_developers(new_grant_name)
+    list_schemas_page = developers_page.click_manage_schemas(grant_name=new_grant_name)
```

Comment:
Having these assertions logically separated in different methods or something (from clicking through the form runner pages) feels like it might make this easier to understand what bits of the journey its doing from a distance? I'm not sure if that works nicely or if it would introduce complications as I haven't been involved in the e2e test layout conversations but certainly isn't a deal breaker!